ogc/query-with-cql-filters/src/main/java/com/esri/samples/query_with_cql_filters/QueryWithCqlFiltersController.java

```diff
@@ -0,0 +1,316 @@
+          setText(item);
+        }
+      });
+
+    } catch (Exception exception) {
+      exception.printStackTrace();
+      new Alert(Alert.AlertType.ERROR, exception.getMessage()).show();
+    }
+  }
+
+  /**
+   * Populates features from provided query parameters, and displays the result on the map.
+   */
+  @FXML
+  private void query() {
+
+    QueryParameters cqlQueryParameters = new QueryParameters();
+    // set the query parameter's where clause with the CQL query in the combo box
+    cqlQueryParameters.setWhereClause(comboBox.getSelectionModel().getSelectedItem());
+    // set the max features to the number entered in the text field
+    cqlQueryParameters.setMaxFeatures(Integer.parseInt(textField.getText()));
+
+    // if the time extent checkbox is selected, retrieve the date selected from the date picker and set it to the
+    // query parameters time extent
+    if (timeExtentCheckBox.isSelected()) {
+      // get the value selected from the date picker
+      LocalDate startDate = startDatePicker.getValue();
+      LocalDate endDate = endDatePicker.getValue();
+
+      // convert the value into a calendar
+      Calendar start = new Calendar.Builder().setDate(startDate.getYear(), startDate.getMonthValue(), startDate.getDayOfMonth()).build();
+      Calendar end = new Calendar.Builder().setDate(endDate.getYear(), endDate.getMonthValue(), endDate.getDayOfMonth()).build();
+
+      // set the query parameters time extent
+      cqlQueryParameters.setTimeExtent(new TimeExtent(start, end));
+    }
+
+    // populate and load the table with the query parameters
+    // set the clearCache parameter to true to clear existing table entries and set the outfields parameter to null requests all fields
+    ListenableFuture<FeatureQueryResult> result = ogcFeatureCollectionTable.populateFromServiceAsync(cqlQueryParameters, true, null);
+    result.addDoneListener(() -> {
+
+      // display number of features returned
+      setFeatureNumberLabelText();
+
+      try {
+        // create a new list to store returned geometries in
+        List<Geometry> featureGeometryList = new ArrayList<>();
+
+        // iterate through each result to get its geometry and add it to the geometry list
+        result.get().iterator().forEachRemaining(feature -> {
+          featureGeometryList.add(feature.getGeometry());
+          feature.getGeometry();
+        });
+
+        if (!featureGeometryList.isEmpty()) {
+          // zoom to the total extent of the geometries returned by the query
+          var totalExtent = GeometryEngine.combineExtents(featureGeometryList);
+          mapView.setViewpointGeometryAsync(totalExtent, 20);
+        }
+
+      } catch (InterruptedException | ExecutionException e) {
+        e.printStackTrace();
+      }
+
+      revertToInitialQueryButton.setDisable(false);
+
+    });
+
+  }
+
+  /**
+   * Handles user interaction with the date picker to ensure the end date time is not earlier than the start date.
```

Comment:
looked into this: there is a bind property but it requires more logic to set it up to unbind it again it would seem after the bind has been set. it would add more lines of code than what is already here. let me know if you've seen something that could work!